ParabolInc/parabol
# fix(metrics): fix the bug where user deletion event won't update HubSpot

Fixes #4692 

Couple of issues:
1. When executing GraphQL query, we should use `ServerAuthToken` rather than using user's token as we need admin access to run the private schema
2. `isRemoved` and company `userCount` & `activeUserCount` will only be accurate after the deletion is done so I moved the HubSpot logic accordingly
3. After deletion, the user email will be augmented. To match with the record in HubSpot (it uses email), we manually set the field to user's email before deletion.

Tests:
1. Copy the [`HUBSPOT_API_KEY`](https://github.com/ParabolInc/action-devops/blob/master/environments/production#L67) to your `.env`
4. Register a test account
5. Delete the account from User->Profile
6. Go to [HubSpot](https://app.hubspot.com/contacts/3888472/contact/76428101), find the test account you created and verify "Is User Removed?" is true.
<img width="306" alt="Screen Shot 2022-05-10 at 3 38 22 PM" src="https://user-images.githubusercontent.com/1879975/167733067-6168a571-fe85-4d1c-a142-e5d84ef240b9.png">
